src/mod_register.erl

```diff
@@ -438,51 +438,24 @@ check_timeout(Source) ->
     Timeout = mongoose_config:get_opt(registration_timeout),
     case is_integer(Timeout) of
         true ->
-            Priority = -(erlang:system_time(second)),
-            CleanPriority = Priority + Timeout,
-            F = fun() -> check_and_store_ip_entry(Source, Priority, CleanPriority) end,
-
-            case mnesia:transaction(F) of
-                {atomic, Res} ->
-                    Res;
-                {aborted, Reason} ->
-                    ?LOG_ERROR(#{what => reg_check_timeout_failed,
-                                 reg_source => Source, reason => Reason}),
-                    true
-            end;
+            TS = erlang:system_time(second),
+            clean_ets(TS - Timeout),
```

Comment:
My only concern here is that we traverse the whole table for every single registration. It should be good enough until a long timeout is used and users are registering quickly.

Reply:
It was like that already in a way, every process would traverse the entire treap and then atomically replace it. In a way there's more lock contention now, but way less garbage. Nevertheless properly cleaning this thing is a bit out of the scope of this ticket I think, and also registration is something that should seldom be enabled publicly so we shouldn't prioritise much optimising it too much 🤔

Reply:
After discussion we decided to use the simples solution for this PR.